pkg/controllers/machinesetsync/machineset_sync_controller.go

```diff
@@ -98,88 +105,218 @@ func (r *MachineSetSyncReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return nil
 }
 
-// Reconcile reconciles CAPI and MAPI machines for their respective namespaces.
-//
-//nolint:funlen
+// Reconcile reconciles CAPI and MAPI MachineSets for their respective namespaces.
 func (r *MachineSetSyncReconciler) Reconcile(ctx context.Context, req reconcile.Request) (ctrl.Result, error) {
-	logger := log.FromContext(ctx, "namespace", req.Namespace, "name", req.Name)
+	logger := log.FromContext(ctx).WithValues("namespace", req.Namespace, "name", req.Name)
+	ctx = logr.NewContext(ctx, logger)
 
-	logger.V(1).Info("Reconciling machineset")
-	defer logger.V(1).Info("Finished reconciling machineset")
+	logger.V(1).Info("Reconciling machine set")
+	defer logger.V(1).Info("Finished reconciling machine set")
+
+	mapiMachineSet, capiMachineSet, err := r.fetchMachineSets(ctx, req.Name)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to fetch machine sets: %w", err)
+	}
 
-	var mapiMachineSetNotFound, capiMachineSetNotFound bool
+	if mapiMachineSet == nil && capiMachineSet == nil {
+		logger.Info("Both MAPI and CAPI machine sets not found, nothing to do")
+		return ctrl.Result{}, nil
+	}
 
-	// Get the MAPI MachineSet.
-	mapiMachineSet := &machinev1beta1.MachineSet{}
-	mapiNamespacedName := client.ObjectKey{
-		Namespace: r.MAPINamespace,
-		Name:      req.Name,
+	if mapiMachineSet == nil {
+		logger.Info("Only CAPI machine set found, nothing to do")
+		return ctrl.Result{}, nil
 	}
 
-	if err := r.Get(ctx, mapiNamespacedName, mapiMachineSet); apierrors.IsNotFound(err) {
-		logger.Info("MAPI MachineSet not found")
+	return r.syncMachineSets(ctx, mapiMachineSet, capiMachineSet)
+}
+
+// fetchMachineSets fetches both MAPI and CAPI MachineSets.
+func (r *MachineSetSyncReconciler) fetchMachineSets(ctx context.Context, name string) (*machinev1beta1.MachineSet, *capiv1beta1.MachineSet, error) {
+	logger := log.FromContext(ctx)
+
+	// Fetch MAPI MachineSet
+	mapiMachineSet := &machinev1beta1.MachineSet{}
+
+	if err := r.Get(ctx, client.ObjectKey{Namespace: r.MAPINamespace, Name: name}, mapiMachineSet); apierrors.IsNotFound(err) {
+		logger.Info("MAPI machine set not found", "name", name)
 
-		mapiMachineSetNotFound = true
+		mapiMachineSet = nil
 	} else if err != nil {
-		logger.Error(err, "Failed to get MAPI MachineSet")
-		return ctrl.Result{}, fmt.Errorf("failed to get MAPI MachineSet: %w", err)
+		logger.Error(err, "Failed to get MAPI machine set", "name", name)
```

Comment:
Hmm - I don't have a strongly held opinion here. 

I think in this case it could be redundant, although the log line will contain all of the k/v pairs we add in context to get to that failure.